app/src/main/java/org/fossasia/openevent/general/sponsor/SponsorWithEventDao.kt

```diff
@@ -9,11 +9,10 @@ import androidx.room.Query
 @Dao
 interface SponsorWithEventDao {
     @Query("""
-        SELECT * FROM sponsor
-        INNER JOIN sponsorwithevent ON
-        sponsor_id = sponsorwithevent.sponsor_id
-        WHERE sponsorwithevent.event_id = :eventID
-        """)
+        SELECT * FROM sponsor s
+        WHERE EXISTS (SELECT * FROM sponsorwithevent se
+        WHERE s.id = se.sponsor_id AND se.event_id = :eventID)
```

Comment:
This is suboptimal. Please revert db changes from this PR. It should be focused on Koin only

Reply:
it's also a problem in the travis build where the query used before contains excessive columns

Reply:
Please paste the error here

Reply:
```
w: /home/travis/build/fossasia/open-event-android/app/build/tmp/kapt3/stubs/fdroidDebug/org/fossasia/openevent/general/speakers/SpeakerWithEventDao.java:11: warning: The query returns some columns [event_id, speaker_id] which are not used by org.fossasia.openevent.general.speakers.Speaker. You can use @ColumnInfo annotation on the fields to specify the mapping.  You can suppress this warning by annotating the method with @SuppressWarnings(RoomWarnings.CURSOR_MISMATCH). Columns returned by the query: id, name, email, photoUrl, shortBiography, longBiography, speakingExperience, position, mobile, location, country, city, organisation, gender, website, twitter, facebook, linkedin, github, isFeatured, event_id, speaker_id. Fields in org.fossasia.openevent.general.speakers.Speaker: id, name, email, photoUrl, shortBiography, longBiography, speakingExperience, position, mobile, location, country, city, organisation, gender, website, twitter, facebook, linkedin, github, isFeatured.
    public abstract androidx.lifecycle.LiveData<java.util.List<org.fossasia.openevent.general.speakers.Speaker>> getSpeakerWithEventId(long eventID);
```

Reply:
```
w: /home/travis/build/fossasia/open-event-android/app/build/tmp/kapt3/stubs/fdroidDebug/org/fossasia/openevent/general/sponsor/SponsorWithEventDao.java:11: warning: The query returns some columns [event_id, sponsor_id] which are not used by org.fossasia.openevent.general.sponsor.Sponsor. You can use @ColumnInfo annotation on the fields to specify the mapping.  You can suppress this warning by annotating the method with @SuppressWarnings(RoomWarnings.CURSOR_MISMATCH). Columns returned by the query: id, name, description, url, logoUrl, level, type, event_id, sponsor_id. Fields in org.fossasia.openevent.general.sponsor.Sponsor: id, name, description, url, logoUrl, level, type.
    public abstract androidx.lifecycle.LiveData<java.util.List<org.fossasia.openevent.general.sponsor.Sponsor>> getSponsorWithEventId(long eventID);
```

Reply:
These are just warnings. Will tackle them later and in a separate issue

Reply:
I'll make a new issue now